test/integration/ClientProxyTest.js

```diff
@@ -43,26 +46,63 @@ describe('ClientProxyTest', function () {
     });
 
     it('client without active connection should return unknown version', function () {
-        const connectionManagerStub = sandbox.stub(ClientConnectionManager.prototype);
-        connectionManagerStub.getActiveConnections.returns({});
+        const connectionRegistryStub = sandbox.stub(ConnectionRegistryImpl.prototype);
+        connectionRegistryStub.getConnections.returns({});
+
+        const proxyManagerStub = sandbox.stub(ProxyManager.prototype);
+
+        const loggingServiceStub = sandbox.stub(LoggingService.prototype);
+        const loggerStub = sandbox.stub(ILogger);
+        loggingServiceStub.getLogger.returns(loggerStub);
+
         const clientStub = sandbox.stub(Client.prototype);
-        clientStub.getConnectionManager.returns(connectionManagerStub);
+        clientStub.getLoggingService.returns(loggingServiceStub);
 
-        const mapProxy = new MapProxy(clientStub, 'mockMapService', 'mockMap');
+        const mapProxy = new MapProxy(
+            'mockMapService',
+            'mockMap',
+            proxyManagerStub,
+            clientStub.getPartitionService(),
+            clientStub.getInvocationService(),
+            clientStub.getSerializationService(),
+            clientStub.getListenerService(),
+            clientStub.getClusterService(),
+            connectionRegistryStub
+        );
         assert.equal(mapProxy.getConnectedServerVersion(), -1);
     });
 
     it('client with a 4.1 server connection should return the version', function () {
+
         const connectionStub = sandbox.stub(ClientConnection.prototype);
         connectionStub.getConnectedServerVersion.returns('40100');
-        const connectionManagerStub = sandbox.stub(ClientConnectionManager.prototype);
-        connectionManagerStub.getActiveConnections.returns({
+
+        const connectionRegistryStub = sandbox.stub(ConnectionRegistryImpl.prototype);
+        connectionRegistryStub.getConnections.returns({
             'localhost': connectionStub
```

Comment:
what about this?

Reply:
Yes, this one should be an array. I wonder why I didn't notice that before.

Reply:
Also, it's not `Array.from(map)`, it's `Array.from(map.values())`

Reply:
You are right, I should change these in next refactor then. So the change will be like

```python
connectionRegistryStub.getConnections.returns([connectionStub])
```